### DISCUSSION TITLE
How do I get the examples to run?

### DISCUSSION THREAD

**Discussion by nilsdoehring** [FlutterGL/flutter_web_gl#13] [Q&A]
Hi and thank you for this great project!

I am wondering: Is the example supposed to work on emulated devices?
I'm using the latest stable releases of Android Studio, Flutter, Dart on an intel Mac, and I'm running into the following issues:

- Android emulator (API 30) results in `RenderingContext.bufferData GLES error 1280` in cube.dart line 60.
- iOS simulator (IOS 14.4) build fails with error `Undefined symbols for architecture x86_64` (details below).

Any idea?

iOS simulator build error:
> Undefined symbols for architecture x86_64:
>       "_eglCreateWindowSurface", referenced from:
>           -[FlutterWebGlPlugin handleMethodCall:result:] in FlutterWebGlPlugin.o
>       "_eglChooseConfig", referenced from:
>           -[FlutterWebGlPlugin handleMethodCall:result:] in FlutterWebGlPlugin.o
>       "_eglMakeCurrent", referenced from:
>           -[FlutterWebGlPlugin handleMethodCall:result:] in FlutterWebGlPlugin.o
>       "_eglGetError", referenced from:
>           -[FlutterWebGlPlugin handleMethodCall:result:] in FlutterWebGlPlugin.o
>       "_eglInitialize", referenced from:
>           -[FlutterWebGlPlugin handleMethodCall:result:] in FlutterWebGlPlugin.o
>       "_eglCreateImageKHR", referenced from:
>           -[FlutterGlTexture createMtlTextureFromCVPixBufferWithWidth:andHeight:] in FlutterWebGlPlugin.o
>       "_eglQueryDisplayAttribEXT", referenced from:
>           _GetANGLEMtlDevice in FlutterWebGlPlugin.o
>       "_eglQueryDeviceAttribEXT", referenced from:
>           _GetANGLEMtlDevice in FlutterWebGlPlugin.o
>       "_eglGetDisplay", referenced from:
>           -[FlutterGlTexture createMtlTextureFromCVPixBufferWithWidth:andHeight:] in FlutterWebGlPlugin.o
>           -[FlutterWebGlPlugin handleMethodCall:result:] in FlutterWebGlPlugin.o
>       "_eglGetProcAddress", referenced from:
>           -[FlutterGlTexture createMtlTextureFromCVPixBufferWithWidth:andHeight:] in FlutterWebGlPlugin.o
>     ld: symbol(s) not found for architecture x86_64
>     clang: error: linker command failed with exit code 1 (use -v to see invocation)
>     note: Using new build system
>     note: Building targets in parallel
>     note: Planning build
>     note: Constructing build description
> 
> Could not build the application for the simulator.
> Error launching application on iPhone 12 Pro Max.

Android emulator runtime error:

> Syncing files to device Android SDK built for x86...
> D/EGL_emulation( 4441): eglCreateContext: 0xf42e2bb0: maj 3 min 0 rcv 3
> D/EGL_emulation( 4441): eglMakeCurrent: 0xf42e2bb0: ver 3 0 (tinfo 0xf4638970) (first time)
> I/FlutterWegGL( 4441): OpenGL initialized: Vendor:Google (ATI Technologies Inc.)ANDROID_EMU_read_color_buffer_dma  renderer: Android Emulator OpenGL ES Translator (AMD Radeon Pro 555X OpenGL Engine)ANDROID_EMU_read_color_buffer_dma  Version: OpenGL ES 3.0 (4.1 ATI-4.6.20)ANDROID_EMU_read_color_buffer_dma 
> D/OpenGLManager( 4441): Successfully initialized GL in plugin
> W/Gralloc4( 4441): allocator 3.x is not supported
> I/flutter ( 4441): EGL version: 1.4
> D/HostConnection( 4441): HostConnection::get() New Host Connection established 0xf42e2980, tid 4470
> D/HostConnection( 4441): HostComposition ext ANDROID_EMU_CHECKSUM_HELPER_v1 ANDROID_EMU_native_sync_v2 ANDROID_EMU_native_sync_v3 ANDROID_EMU_native_sync_v4 ANDROID_EMU_dma_v1 ANDROID_EMU_direct_mem ANDROID_EMU_host_composition_v1 ANDROID_EMU_host_composition_v2 ANDROID_EMU_YUV_Cache ANDROID_EMU_async_unmap_buffer ANDROID_EMU_has_shared_slots_host_memory_allocator ANDROID_EMU_sync_buffer_data ANDROID_EMU_read_color_buffer_dma GL_OES_EGL_image_external_essl3 GL_OES_vertex_array_object GL_KHR_texture_compression_astc_ldr ANDROID_EMU_host_side_tracing ANDROID_EMU_async_frame_commands ANDROID_EMU_gles_max_version_3_0 
> D/EGL_emulation( 4441): eglCreateContext: 0xf42e2a60: maj 3 min 0 rcv 3
> D/EGL_emulation( 4441): eglMakeCurrent: 0xf42e2a60: ver 3 0 (tinfo 0xc0fc8cf0) (first time)
> I/FlutterWebGL( 4441): Created a new texture 640x320
> I/flutter ( 4441): RenderingContext.bufferData GLES error 1280

**Comment by escamoteur** [comment]
Hi, sorry for the long time to respond for me, but I had to struggle with my mental health for the last 7 months.

you have to make sure your Emulator uses the latest EGL Version. you can set that in the Advanced setting of your Emulator. 
On iOS this is indeed a problem that I couldn't solve yet
